### ISSUE TITLE
BackgroundService should provide a linked token to ExecuteAsync

### ISSUE THREAD

**Issue by klettier** [dotnet/docs#16456]
In BackgroundService the token provided to ExecuteAsync should be linked (CancellationTokenSource.CreateLinkedTokenSource) to the one provided to StartAsync, so _stoppingCts could be cancelled by both the caller of StartAsync and BackgroundService.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d49a03c0-a844-26eb-48a5-33a612dd3ead
* Version Independent ID: 0707865f-9db7-0d71-42a5-bc1a1e89680a
* Content: [Implement background tasks in microservices with IHostedService and the BackgroundService class](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice)
* Content Source: [docs/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md](https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md)
* Product: **dotnet-architecture**
* Technology: **microservices**
* GitHub Login: @nishanil
* Microsoft Alias: **nanil**

**Comment by klettier** [comment]
Linked to https://github.com/aspnet/Extensions/issues/2811

**Comment by mvelosop** [comment]
Hi @klettier, thanks for the feedback!

I've been reading this issue and related PR, or quite a while, including the related issue [Confusing Cancellation in BackgroundService (dotnet/extensions#1245)](https://github.com/dotnet/extensions/issues/1245), and find the whole topic quite interesting.

However, from issue https://github.com/dotnet/extensions/issues/1245 the only thing that's clear to me is that the semantics for the cancellation token in `StartAsync()` is confusing. Even though I tend to go along @davidfowl's view, and find, IMHO, [the source comment quite clear](https://github.com/dotnet/extensions/blob/5de6ef11c256f3caeb443b331411d620fb113a98/src/Hosting/Abstractions/src/IHostedService.cs#L15-L19):

```csharp
/// <summary>
/// Triggered when the application host is ready to start the service.
/// </summary>
/// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
        Task StartAsync(CancellationToken cancellationToken);
```

Anyway, I wouldn't really update this doc just yet, because the related PR, even though merged, is for a yet unreleased version (5.0-preview), so it doesn't really apply for 3.1.

Hope this helps.

cc/ @nishanil, @tdykstra

**Comment by klettier** [comment]
Thx for your response, do I have to close the issue of do you need it for the **5.0-preview** doc ?
